### PR TITLE
feat: Add Create function for githubipallowlist_ip_allow_list_entry resource

### PR DESCRIPTION
**Pull request by pzmi-f3**
This change adds a `githubipallowlist_ip_allow_list_entry` resource with a limited implementation. It implements one of the functions managing `githubipallowlist_ip_allow_list_entry`, Create, on an organization level. The resource still lacks Read, Update and Delete functions, so it's not fully functional. Intention of this change is to provide a chunk of code that's manageable for a reviewer together that can actually be run with a real results (changes to the allow list).

To validate changes run the `simple` example from `examples` directory:

1. Follow [Building The Provider section from README](https://github.com/form3tech-oss/terraform-provider-githubipallowlist/blob/5f6a0e2f6bafc5c041256b8fd4512eb8a0b7e936/README.md#building-the-provider)
2. Generate a Personal Access Token (classic) with an `admin:org` scope.
3. Run the following script filling missing variables:

  ```shell
  cd $PROJECT_ROOT/examples/simple
  export GITHUB_TOKEN=$GENERATED_TOKEN
  export GITHUB_ORGANIZATION=$YOUR_ORGANIZATION_NAME
  terraform plan
  terraform apply
  ```

4. Manually delete IP `1.2.3.4/32` allow list entry from your organization.

Example output of `terraform apply`:

image

### CODE REVIEW

File: github/client.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+	}
+	return res, nil
+}
+
+func handleGraphQLResponse(res *http.Response) (*GraphQLResponse, error) {
+	defer res.Body.Close()
+
+	resBytes, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "response body read error")
```
Comment (bart-lisiecki-form3) [inline github/client.go:211]: nit: We could probably return `ErrorWithStatusCode` here, might be useful for debugging

File: github/client.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+		return nil, ErrorWithStatusCode{
+			StatusCode: res.StatusCode,
+			message:    fmt.Sprintf("GitHub API response: %s", string(resBytes)),
+		}
+	}
+
+	gqlRes := new(GraphQLResponse)
+	err = json.Unmarshal(resBytes, gqlRes)
+	if err != nil {
+		return nil, errors.Wrap(err, "response unmarshalling error")
```
Comment (bart-lisiecki-form3) [inline github/client.go:224]: nit: Is there a reason we don't want to return `ErrorWithStatusCode` here as well?

**Review by bart-lisiecki-form3 — APPROVED** [review]
Looks good overall!

There was a lot of additions, so I hope I didn't miss anything. I appreciate that you reduced the amount of changes to minimum, it's not always easy to introduce chunks that are small but still meaningful!

File: github/client.go (hunk 2 above)
Comment (pzmi-f3) [reply github/client.go:224]: I missed that, thanks

**Review by bart-lisiecki-form3 — APPROVED** [review]

**Pull request merged**